haystack/modeling/model/multimodal/__init__.py

```diff
@@ -0,0 +1,198 @@
+    model_type = ""
+    model_wrapper_class: Type[HaystackModel]
+
+    # Prepare the kwargs the model wrapper expects (see each wrapper's init for details)
+    wrapper_kwarg_groups = {}
+    wrapper_kwarg_groups["model_kwargs"] = model_kwargs
+
+    # SentenceTransformers are much faster, so use them whenever possible
+    if _is_sentence_transformers_model(
+        pretrained_model_name_or_path, use_auth_token=autoconfig_kwargs.get("use_auth_token", False)
+    ):
+        model_wrapper_class = HaystackSentenceTransformerModel
+        try:
+            # Use AutoConfig to log some more info about the model class
+            config = AutoConfig.from_pretrained(pretrained_model_name_or_path=model_name, **autoconfig_kwargs)
+            model_type = config.model_type
+        except Exception as e:
+            logger.debug(f"Can't find model type for {pretrained_model_name_or_path}: {e}")
+
+        if feature_extractor_kwargs is not None:
+            logger.debug(
+                "Can't forward feature_extractor_kwargs to a SentenceTransformers model. "
+                "These kwargs are being dropped. "
+                f"Content of feature_extractor_kwargs: {feature_extractor_kwargs}"
+            )
+
+    else:
+        # Use AutoConfig to understand the model class
+        config = AutoConfig.from_pretrained(pretrained_model_name_or_path=model_name, **autoconfig_kwargs)
+        if not config.model_type:
+            logger.error(
+                f"Model type not understood for '{pretrained_model_name_or_path}'. Please provide the name of "
+                "a model that can be downloaded from the Model Hub.\nUsing the AutoModel class. "
+                "THIS CAN CAUSE CRASHES and won't work for models that are not working with text."
+            )
+            model_type = "AutoModel"
+        else:
+            try:
+                model_type = HUGGINGFACE_CAPITALIZE[config.model_type.lower()]
+            except KeyError as e:
+                logger.error(
+                    f"Model '{pretrained_model_name_or_path}' (type '{config.model_type.lower()}') "
+                    "is not supported by Haystack. Using the AutoModel class for it. "
+                    "THIS CAN CAUSE CRASHES and won't work for models that are not working with text. "
+                    f"Supported model types: {', '.join(HUGGINGFACE_CAPITALIZE.keys())}"
+                )
+                model_type = "AutoModel"
+
+        # Find the HF class corresponding to this model type
+        try:
+            model_wrapper_class = HUGGINGFACE_TO_HAYSTACK[model_type]
+        except KeyError as e:
+            raise ValueError(
+                f"The type of the given model (name/path: {pretrained_model_name_or_path}, detected type: {model_type}) "
+                "is not supported by Haystack or was not correctly identified. Please use supported models only. "
+                f"Supported model types: {', '.join(HUGGINGFACE_TO_HAYSTACK.keys())}"
```

Comment:
Related to https://github.com/deepset-ai/haystack/pull/2891/files#r993155924
```suggestion
                f"Haystack doesn't support the model '{pretrained_model_name_or_path}' of type '{model_type}' "
                "or failed to identify its type. If the type was wrongly identified, please open an issue [on GitHub](https://github.com/deepset-ai/haystack/issues). "
                f"Here's a list of model types we support: {', '.join(HUGGINGFACE_TO_HAYSTACK.keys())}"
```